lb/roundrobin.go

```diff
@@ -111,11 +112,11 @@ func (s *ResultCache) getNextResult(key string) (net.SRV, error) {
 	}
 
 	if len(srvs) <= i {
-		i = 0
+		i = i % len(srvs)
 	}
 
 	srv = srvs[i]
-	s.m[key].i += 1
+	s.m[key].i = i + 1
```

Comment:
The difference is here @benschw , original code set `i` to `s.m[key].i` which leads to a situation that after `i` reaches the largest index, `srv` will only get the first value in the array.